frb_example/benches/dart/lib/interceptor.dart

```diff
@@ -0,0 +1,182 @@
+/// interceptor (platform-agnostic wrapper)
+///
+/// record time before and after executing async or sync functions over the bridge.
+///
+/// can also generate
+/// a [continuous-benchmark](https://github.com/marketplace/actions/continuous-benchmark)-compliant json output on demand
+
+import 'dart:async';
+
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+import 'package:logging/logging.dart';
+import 'bridge_definitions.dart';
+import 'package:uuid/uuid.dart';
+import 'interceptor.io.dart' if (dart.library.html) 'interceptor.web.dart';
+import 'bridge_generated.io.dart' if (dart.library.html) 'bridge_generated.web.dart';
+
+/// this is the custom platform, which contains the interceptor.
+abstract class FlutterRustBridgeExampleBenchmarkSuitePlatformBenchBase<
+    STDOUT extends FlutterRustBridgeInterceptorStdOut,
+    JSON extends FlutterRustBridgeInterceptorJson> extends FlutterRustBridgeExampleBenchmarkSuitePlatform {
+  FlutterRustBridgeExampleBenchmarkSuitePlatformBenchBase(ExternalLibrary dylib) : super(dylib);
+  FlutterRustBridgeInterceptor<TimeWatch> get interceptor;
+
+  @override
+  Future<S> executeNormal<S>(FlutterRustBridgeTask<S> task) async {
+    final String debugName = task.constMeta.debugName;
+    final TimeWatch stopwatch = await interceptor.beforeExecuteNormal(debugName, task.hint);
+    final result = await super.executeNormal(task);
+    await interceptor.afterExecuteNormal(debugName, stopwatch);
+    return result;
+  }
+
+  @override
+  S executeSync<S>(FlutterRustBridgeSyncTask task) {
+    final String debugName = task.constMeta.debugName;
+    final TimeWatch stopwatch = interceptor.beforeExecuteSync(debugName, task.hint);
+    final result = super.executeSync(task);
+    interceptor.afterExecuteSync(debugName, stopwatch);
+    return result;
+  }
+
+  Future<List<Metric>?> metrics() async => interceptor is FlutterRustBridgeInterceptorJson
```

Comment:
uber-nit: Btw looks like this do not need Future or async

Reply:
looks like dart linter has a different opinion about it, see:

https://user-images.githubusercontent.com/21016014/194580572-857daf1e-a92e-45c7-80a9-71028847d1cf.mov

Reply:
Also need to remove "async" keyword

Reply:
done in 461e0d3